backend/modules/auth/provider.py

```
@@ -0,0 +1,454 @@
+            if IS_PRODUCTION:
+                logger.warning("⚠️  OAUTH_ENCRYPTION_KEY not found in production - OAuth state encryption disabled. Set OAUTH_ENCRYPTION_KEY for secure OAuth flows.")
+                self.cipher = None
+            else:
+                # Development mode: use or generate persistent key
+                try:
+                    # Go up 3 levels from backend/modules/auth/provider.py -> backend/
+                    dev_key_file = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))), '.oauth_encryption_key')
+                    key = self._get_or_create_dev_key(dev_key_file)
+                    logger.warning("No OAUTH_ENCRYPTION_KEY found, using development key from file. Set OAUTH_ENCRYPTION_KEY in production!")
+                    self.cipher = Fernet(key)
+                except Exception as e:
```

Ruff (0.14.6)

Do not catch blind exception: `Exception`

(BLE001)